src/client.rs

```diff
@@ -48,9 +49,12 @@ enum ReadError {
 
 impl ClientConnection {
     /// Creates a new `ClientConnection` that takes ownership of the `TcpStream`.
+    /// The buffered parameter allows for disabling buffering, which will
+    /// flush the writer after every single write.
     pub fn new(
         write_socket: RefinedTcpStream,
         mut read_socket: RefinedTcpStream,
+        buffered: bool,
```

Comment:
First off, this is a public API and I think we should minimize breakage where possible, so I'd prefer a `new_unbuffered` method rather than changing the signature of `new`.

I'd also prefer an `Enum BufferMode { Buffered, Unbuffered }` or something like that, I feel that boolean behaviour control parameters are an antipattern in Rust where we have such great support for enums.

Reply:
Agreed for the enum and minizing breakage.

However `ClientConnection` does not appear to be public API? I do not see how a client can use it, because it is not reexported publicly in `lib.rs`, and the module is not exposed publicly either.

I will submit a new version shortly, but without a new method for `ClientConnection` as I don't think there will be any API breakage there (and thus any advantage. If you want me to still add a different method, that's not an issue at all, and I will do so gladly.

As for the breakage in `ServerConfig`, I will submit a small change where the buffering option is hidden behind an opaque struct, so that it can be further modified in the future without breaking the API (but that won't prevent that initial API breakage, sadly). Please tell me what you think of that. To be honest, I think that method could also be extended to cover all of `ServerConfig`.

Reply:
The library in general predates `pub(crate)` so there's rather a lot of these mistakes to be made, my apologies if it turns out this is really an internal type that's mistakenly marked `pub`.

I'll re-review your latest version ASAP.